torch_xla/csrc/ir.h

```diff
@@ -110,18 +115,22 @@ class XlaNode : public torch::lazy::Node {
 
   torch::lazy::hash_t node_hash() const { return node_hash_; }
 
-  torch::lazy::hash_t hash() const override { return dag_hash_; }
+  torch::lazy::hash_t hash() const override {
+    if (sharding_hash_ != 0) {
+      return torch::lazy::HashCombine(dag_hash_, sharding_hash_);
```

Comment:
why don't we always combine `dag_hash_` and `sharding_hash_`? From you implementation above `dag_hash_` already include `sharding_hash_`